### PR TITLE
Update outdated URLs in comments

### PR DESCRIPTION
**Pull request by omalyshe**
### Description 
Some URLs have expired and should be replaced.

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [x] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: examples/getting_started/sub_string_finder/README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Sub_string_finder sample
-An example that uses the `parallel_for` template in a substring matching program. The [oneAPI Threading Building Blocks [](https://software.intel.com/content/www/us/en/develop/documentation/get-started-with-onetbb/top.html) describes this example.
+An example that uses the `parallel_for` template in a substring matching program. The [oneAPI Threading Building Blocks [](https://www.intel.com/content/www/us/en/docs/onetbb/get-started-guide/current/overview.html) describes this example.
```
Comment (dnmokhov) [inline examples/getting_started/sub_string_finder/README.md:2]:
```suggestion
An example that uses the `parallel_for` template in a substring matching program. The [oneAPI Threading Building Blocks Get Started Guide](https://www.intel.com/content/www/us/en/docs/onetbb/get-started-guide/current/overview.html) describes this example.
```

**Review by dnmokhov — CHANGES_REQUESTED** [review]

**Review by alexandraepan — DISMISSED** [review]
LGTM!

**Comment by omalyshe** [comment]
@dnmokhov @alexandraepan Please, look. I completed fixes.

**Review by dnmokhov — APPROVED** [review]
LGTM

**Pull request merged**